BookEcommerce/Controllers/VendorController.cs

```diff
@@ -0,0 +1,42 @@
+﻿using BookEcommerce.Models.DTOs;
+using BookEcommerce.Models.DTOs.Response.Base;
+using BookEcommerce.Services.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace BookEcommerce.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+
+    [Authorize(Roles = "VENDOR", AuthenticationSchemes = "Bearer")]
+    public class VendorController : ControllerBase
+    {
+        private readonly IVendorService vendorService;
+        public VendorController(IVendorService vendorService)
+        {
+            this.vendorService = vendorService;
+        }
+        [Authorize(AuthenticationSchemes = "Bearer", Roles = "VENDOR")]
+        [HttpPost("create")]
+        public async Task<IActionResult> CreateProfile([FromForm] CreateVendorDTO CreateVendorDTO)
+        {
+            string AuthHeader = Request.Headers["Authorization"].ToString().Split(' ')[1];
+            Console.WriteLine(AuthHeader);
+            var result = await this.vendorService.CreateVendor(CreateVendorDTO, AuthHeader);
+            if (result.IsSuccess)
+                return StatusCode(StatusCodes.Status200OK, new ResponseBase
```

Comment:
return OK is simply for me. Btw, Status is OK and message is still OK. So why do I need this mess?

Reply:
i have validated email params in service, you can take a little bit look